src/qml/WebPatchPage.qml

```diff
@@ -340,16 +340,17 @@ Page {
                     width: parent.width
                     contentHeight: filesContent.height
                     property bool isInstalled: !!container.versions && container.versions[modelData.project] == modelData.version
-                    property bool isCompatible: modelData.compatible.indexOf(PatchManager.osVersion) >= 0
-                    property bool isReinstallable: isInstalled && isCompatible
+                    property bool isCompatible: (modelData.compatible.indexOf(PatchManager.osVersion) >= 0)
+                    property bool forceCompatible: PatchManager.sfosVersionCheck !== VersionCheck.Strict
+                    property bool isReinstallable: isInstalled && (isCompatible || forceCompatible)
 
                     onClicked: {
-                        if ((PatchManager.sfosVersionCheck !== VersionCheck.Strict) && !isCompatible) {
+                        if (fileDelegate.isReinstallable) {
+                            remorseAction(qsTranslate("", "Re-Install Patch %1").arg(patchData.display_name), installPatch)
+                        } else if (!isCompatible && !forceCompatible) {
```

Comment:
this is equivalent to `!(isCompatible || forceCompatible)` which, when true - and because `isReinstallable` was false because we're on the else branch - tells us nothing about `isInstalled`.
However when `!(isCompatible || forceCompatible)` is false (the next else) this does tells us that `isCompatible || forceCompatible)` is true which tells us that `isInstalled` was the one false failing the `isReinstallable check`.

Therefore re-reading the whole if/else is like:
```
if (isReinstallable) {
  // re-install
} else if (!isCompatible && !forceCompatible) {
  // error 
} else since (isCompatible || forceCompatible) {
  // install
}
```
which looks sound to me.

Before, I think the "only" error was forceCompatible which not tested correctly (it needed a !) 
```
if (forceCompatible && !isCompatible) {
  // error - oops, we are erroring out if forceCompatible is on?
} else since (!forceCompatible || isCompatible) if (!isInstalled) {
  // install - oops we might install if forceCompatible is off?
} else since (!forceCompatible || isCompatible) && isInstalled {
  // re-install we might re-install if forceCompatible is off?l 
}
```

Thanks for making this more readable :)